File: courier_application/app/src/main/java/by/bstu/vs/stpms/courier_application/model/database/dao/AbstractDao.java
package by.bstu.vs.stpms.courier_application.model.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Update;

import java.util.Collection;
import java.util.List;

import by.bstu.vs.stpms.courier_application.model.database.entity.AbstractEntity;

public abstract class AbstractDao<E extends AbstractEntity> {
    @Insert
    public abstract void insert(E item);
    @Delete
    public abstract void delete(E item);
    @Delete
    public abstract void deleteAll(Collection<E> items);
    @Update
    public abstract void update(E item);
    public abstract E findById(long id);
    public abstract List<E> getAll();
}

File: courier_application/app/src/main/java/by/bstu/vs/stpms/courier_application/model/database/dao/AbstractDao.kt
package by.bstu.vs.stpms.courier_application.model.database.dao

import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Update
import by.bstu.vs.stpms.courier_application.model.database.entity.AbstractEntity

abstract class AbstractDao<E : AbstractEntity> {
    @Insert
    abstract fun insert(item: E)
    @Delete
    abstract fun delete(item: E)
    @Delete
    abstract fun deleteAll(items: Collection<E>)
    @Update
    abstract fun update(item: E)
    abstract suspend fun findById(id: Long): E?
    abstract suspend fun getAll(): List<E>
}